Sources/PackageModel/Package.swift

```diff
@@ -137,8 +137,22 @@ public struct PackageReference: JSONMappable, JSONSerializable, Codable, CustomS
 
     /// Compute the default name of a package given its URL.
     public static func computeDefaultName(fromURL url: String) -> String {
+      #if os(Windows)
+        let isSeparator : (Character) -> Bool = { $0 == "/" || $0 == "\\" }
+      #else
+        let isSeparator : (Character) -> Bool = { $0 == "/" }
+      #endif
+       
         // Get the last path component of the URL.
-        var lastComponent = url.split(separator: "/", omittingEmptySubsequences: true).last ?? url[...]
+        // Drop the last character in case it's a trailing slash.
+        var endIndex = url.endIndex
+        if let lastCharacter = url.last, isSeparator(lastCharacter) {
+            endIndex = url.index(before: endIndex)
+        }
+
+        let separatorIndex = url[..<endIndex].lastIndex(where: isSeparator)
```

Comment:
I’m too lazy to fix it because this will probably never be tripped in practice, and because it was just as fragile before this commit. But we should all keep in mind that...

This sort of thing should actually be done on the `unicodeScalars` view of the string. `mkdir -p ́/́/́/́` is a valid (albeit absurd) operation that creates four directories. Yet this code will think the resulting directory only has one path component.